### ISSUE TITLE
Data Browser

### ISSUE THREAD

**Issue by NAnguiano** [SirSeim/SPFY#22]
- [x] UI Improvements
- [x] Enable search of more tables
- [x] Fix search of dates
- [ ] More detailed search

**Comment by NAnguiano** [comment]
I decided to rewrite the entire search portion of the data browser, so it broke the whole thing. In the process of redoing it, I've fixed the search of dates, ints, and booleans, and made them all distinct from each other. But it still isn't fully functional. That should come tomorrow.

Before I scratched the searching portion of the data browser, I got a somewhat more detailed search working. I had it restricted down to two filters (something OR something / something AND Something), but I think it should be more detailed. I was thinking at capping it at 10 different filters, just to keep things from getting too out of hand.

I could use input on the UI - whether it's better, worse, what other design features look good, etc.

What other tables would we want to search? I've refactored the code so that essentially any table should be able to be added with almost no work on my part. I was thinking of adding case notes, but am not sure what others would work well.

Haven't even gotten into Report Generation. I'm going to actually break that out into its own issue, since it's a part of the data browser, but distinct enough that I think it can be branched off.
